### PR TITLE
expose memory key name

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: langchain/chains/conversation/memory.py [hunk 1]
```diff
@@ -230,20 +230,20 @@ class ConversationEntityMemory(Memory, BaseModel):
     llm: BaseLLM
     entity_extraction_prompt: BasePromptTemplate = ENTITY_EXTRACTION_PROMPT
     entity_summarization_prompt: BasePromptTemplate = ENTITY_SUMMARIZATION_PROMPT
-    memory_keys: List[str] = ["entities", "history"]  #: :meta private:
     output_key: Optional[str] = None
     input_key: Optional[str] = None
     store: Dict[str, Optional[str]] = {}
     entity_cache: List[str] = []
     k: int = 3
+    chat_history_key: str = "history"
```
Comment (rowillia) [inline langchain/chains/conversation/memory.py:238]: Is this constant?  If so you can use the `const` parameter from Pydantic:

```python
chat_history_key: str = Field("history", const=True)
```

**Comment by VBoB13** [comment]
While I appreciate the efforts, I still can't use this feature due to the very same issue.
Error:
```python
| ERROR | 2023-03-03 18:03:15 | /api/prompt/prompt.py | PromptMaker Could NOT execute LangChain's agent code!
Details:
Missing some input keys: {'chat_history'}  File "/opt/gpt3/api/prompt/prompt.py", line 134, in _run
    results = agent_chain.run(
  File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 242, in run
    return self(kwargs)[self.output_keys[0]]
  File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 132, in __call__
    inputs = self.prep_inputs(inputs)
  File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 221, in prep_inputs
    self._validate_inputs(inputs)
  File "/usr/local/lib/python3.10/site-packages/langchain/chains/base.py", line 101, in _validate_inputs
    raise ValueError(f"Missing some input keys: {missing_keys}")
```

But my code does indeed specify to look for 'chat_history'; in multiple places even:
```python
prompt = ConversationalAgent.create_prompt(
                    tools,
                    prefix=TEMPLATE_OBJECT[self.language]["prefix"],
                    suffix=TEMPLATE_OBJECT[self.language]["suffix"],
                    input_variables=['input', 'chat_history',
                                     'agent_scratchpad', 'context']
                )
                memory = ConversationBufferMemory(memory_key="chat_history")
                llm_chain = LLMChain(llm=self.llm, prompt=prompt)
                tools = load_tools(["serpapi", "llm-math"], llm=self.llm)
                # ConversationalAgent seems to have issues with chat_history
                # TODO: Check ConversationalAgent + {chat_history} !!!
                # agent = ConversationalAgent(
                #     llm_chain=llm_chain, tools=tools, verbose=True)
                agent = ConversationalAgent(
                    llm_chain=llm_chain, tools=tools, verbose=True)
                agent_chain = AgentExecutor.from_agent_and_tools(
                    agent, tools, verbose=True, mempry=memory)
                results = agent_chain.run(
                    input=self.prompt, context=self.context)
```

Is there anything I am missing? Am I the one being silly? 😢

**Comment by VBoB13** [comment]
Oh my goodness.... It was all my fault in the end.... Spelling mistakes 😭 

```python
agent_chain = AgentExecutor.from_agent_and_tools(
    agent, tools, verbose=True, mempry=memory)
```
-->
```python
agent_chain = AgentExecutor.from_agent_and_tools(
    agent, tools, verbose=True, memory=memory)
```

Thanks for your great work! This tool is freakin' amazeballs :)

**Comment by postix** [comment]
about the chat_history ... how do you handle the sessions with the client ??? using a fingerprint or something to make a history not shared with others users ...

**Pull request merged**